verilog/formatting/formatter_test.cc

```diff
@@ -5190,6 +5190,121 @@ static const std::initializer_list<FormatterTestCase> kFormatterTestCases = {
         "JgQLBG == 4'h0;, \"foo\")\n"
         "endtask\n",
     },
+
+    // module instantiation named ports tabular alignment
+    {// all named ports
+        "module m;\n"
+        "foo bar(.a(a), .aa(aa), .aaa(aaa));\n"
+        "endmodule\n",
+        "module m;\n"
+        "  foo bar (\n"
+        "      .a  (a),\n"
+        "      .aa (aa),\n"
+        "      .aaa(aaa)\n"
+        "  );\n"
+        "endmodule\n"
+    },
+    {// named ports with pre-proc directives
+        "module m;\n"
+        "foo bar(.a(a), `ifdef MACRO .aa(aa), `endif .aaa(aaa));\n"
+        "endmodule\n",
+        "module m;\n"
+        "  foo bar (\n"
+        "      .a  (a),\n"
+        "`ifdef MACRO\n"
+        "      .aa (aa),\n"
+        "`endif\n"
+        "      .aaa(aaa)\n"
+        "  );\n"
+        "endmodule\n"
+    },
+    {// named ports with macros
+        "module m;\n"
+        "foo bar(.a(a), .aa(aa[`RANGE]), .aaa(aaa));\n"
+        "endmodule\n",
+        "module m;\n"
+        "  foo bar (\n"
+        "      .a  (a),\n"
+        "      .aa (aa[`RANGE]),\n"
+        "      .aaa(aaa)\n"
+        "  );\n"
+        "endmodule\n"
+    },
+    {
+        "module m;\n"
+        "foo bar(.a(a), .AA, .aaa(aaa));\n"
+        "endmodule\n",
+        "module m;\n"
+        "  foo bar (\n"
+        "      .a  (a),\n"
+        "      .AA,\n"
+        "      .aaa(aaa)\n"
+        "  );\n"
+        "endmodule\n"
+    },
+    {// name ports with comments
+        "module m;\n"
+        "foo bar(.a(a), .aa(aa)/*comment*/, .aaa(aaa));\n"
+        "endmodule\n",
+        "module m;\n"
+        "  foo bar (\n"
+        "      .a  (a),\n"
+        "      .aa (aa)  /*comment*/,\n"
+        "      .aaa(aaa)\n"
+        "  );\n"
+        "endmodule\n"
+    },
+    {
+        "module m;\n"
+        "foo bar(.a(a),\n"
+        " //.aa(aa),\n" 
+        ".aaa(aaa));\n"
+        "endmodule\n",
+        "module m;\n"
+        "  foo bar (\n"
+        "      .a  (a),\n"
+        "      //.aa(aa),\n"
+        "      .aaa(aaa)\n"
+        "  );\n"
+        "endmodule\n"
+    },
```

Comment:
more suggested test cases:

* An instance with only implicit `.xyz` connections.
* An instance with empty ports, but containing only comments.  The formatted result should at least indent the comments correctly.
* An instance with one or more port with a trailing //comment, e.g. `.aa(aa),  // this is aa`
* An instance with a blank line separating two groups of named ports.
* A port connection that includes a concatenation, like `.p({q,r,s})`
* A port connection that includes a (non-macro) subrange slice, like `.p(q[r:s])`

Reply:
> - An instance with empty ports, but containing only comments. The formatted result should at least indent the comments correctly.

This doesn't really test the kPortActualList tabular alignment. But I tried two versions:
```systemverilog
module m;
foo bar(
   //comment1
//comment2
);
endmodule
```
Gets formatted to:
```systemverilog
module m;
  foo bar (
  //comment1
  //comment2
  );
endmodule
```
I feel that it should be?
```systemverilog
module m;
  foo bar (
      //comment1
      //comment2
  );
endmodule
```

And the following doesn't get formatted at all. But this feels like a different issue.
```systemverilog
module m;
  foo bar( //comment1
  //comment2
  );
endmodule
```

Reply:
Ok, we'll address this later, I thought/hoped it might already be working with your changes.